main.go

```diff
@@ -16,8 +18,27 @@ var yggdDispatchSocketAddr string
 var yggdHandler string
 
 func main() {
-	// Get initialization values from the environment.
 	var ok bool
+
+	configFile, ok := os.LookupEnv("CONFIG_FILE")
+	if ok {
+		config, err := toml.LoadFile(configFile)
+		if err != nil {
+			fmt.Errorf("cannot load config: %w", err)
+			return
+		}
+
+		for _, value := range config.GetArray("env").([]string) {
+			split := strings.Split(value, "=")
+			os.Setenv(split[0], split[1])
+		}
+
+		os.Setenv("YGG_FORWARDER_HANDLER", config.Get("directive").(string))
+	} else {
+		log.Fatal("CONFIG_FILE not set")
+	}
```

Comment:
Can we reuse the original function from the [yggdrasil package](https://github.com/RedHatInsights/yggdrasil/blob/cdde836c519ac72302714b5f7ae90e45de7c79a7/cmd/yggd/worker.go#L22-L45)? 

At least we should copy the code from there. I think it will be better than developing a new paring algo.

BTW, I see that the original code extracts the directive differently (from the config file name). I think we should stick with that.

Reply:
> Can we reuse the original function from the [yggdrasil package](https://github.com/RedHatInsights/yggdrasil/blob/cdde836c519ac72302714b5f7ae90e45de7c79a7/cmd/yggd/worker.go#L22-L45)?

I don't see this code in 0.2.0 and part of why I opted to implement it ourselves. Keep the coupling to minimum for this one-off thing.

> 
> At least we should copy the code from there. I think it will be better than developing a new paring algo.
> 

Funny thing is, I actually started with that code, but it was not working, getting messy and unwieldy. So I read the docs on the `toml` package we are using and hence the solution I have.

> BTW, I see that the original code extracts the directive differently (from the config file name). I think we should stick with that.

Reply:
So let's leave it as is.
The `directive` handling needs to change though.

Reply:
> The `directive` handling needs to change though.

I don't follow this.

Reply:
Yggdrasil is [deriving](https://github.com/RedHatInsights/yggdrasil/blob/cdde836c519ac72302714b5f7ae90e45de7c79a7/cmd/yggd/worker.go#L42) the directive from the configuration file name.
We need to do the same, as opposed to reading it directly from the config file (`config.Get("directive").(string)`)

Reply:
Ah ha! Got it thanks